### ISSUE TITLE
Reduce code-duplication in regionprops

### ISSUE THREAD

**Issue by haesleinhuepf** [morphometrics/morphometrics#37]
Hi @kevinyamauchi ,

I detected some code-duplication between regionprops here:
https://github.com/morphometrics/morphometrics/blob/main/src/morphometrics/measure/label.py#L37-L129

and here:
https://github.com/haesleinhuepf/napari-skimage-regionprops/blob/master/napari_skimage_regionprops/_regionprops.py#L53-L144

Potentially the code on morphometrics side is outdated. As napari-skimage-regionprops now shows up in the widget of morphometrics (since version [0.5.4](https://github.com/haesleinhuepf/napari-skimage-regionprops/releases/tag/0.5.4)), you could remove the code from the morphometrics repository. If there were any fixes in the code on morphometrics side, please send a PR to napari-skimage-regionprops.

Best,
Robert

**Comment by kevinyamauchi** [comment]
Hey! Thank you for reporting this. Super cool that you've made `napari-skimage-regionprops` compatible with `morphometrics` measurements. Indeed, the duplicated code in `morphometrics` was really just intended to prototype the measurement engine and meant to be deleted. I will remove the duplicate code and add `napari-skimage-regionprops` as a dependency :)

On a related note - I have been thinking that it makes sense to separate the "measurement engine" out into a separate package so that people don't need to bring along the "heavier dependencies if they don't want to (e.g., openCL), which can cause challenges installing via pypi (e.g., #35 ) What do you think?

**Comment by haesleinhuepf** [comment]
> On a related note - I have been thinking that it makes sense to separate the "measurement engine" out into a separate package so that people don't need to bring along the "heavier dependencies if they don't want to (e.g., openCL), which can cause challenges installing via pypi (e.g., https://github.com/morphometrics/morphometrics/issues/35 ) What do you think?

Yes, fully agreed! You want to have a minimal-dependency backend for `register_measurement_set` and potentially multiple front-ends. See also: https://github.com/BioimageAnalysisCoreWEHI/napari_lattice/issues/17#issuecomment-1259099402

**Comment by pr4deepr** [comment]
Thanks for tagging me here @haesleinhuepf. 
I would love to have a function that is agnostic to the measurement function being used, and takes a single/multiple images as input. When using `register_measurement_set`, I'm assuming the aim is to make it agnostic of the measurement function used, right?
